command/agent/job_endpoint.go

```diff
@@ -1385,6 +1394,12 @@ func ApiCSIPluginConfigToStructsCSIPluginConfig(apiConfig *api.TaskCSIPluginConf
 	return sc
 }
 
+func ApiActionToStructsAction(job *structs.Job, action *api.Action, act *structs.Action) {
+	act.Name = action.Name
+	act.Args = action.Args
```

Comment:
`Args` isn't a primitive so you'll need to copy this slice too:

```suggestion
	act.Args = slices.Clone(action.Args)
```